File: CS50 problem sets/pset6/readability/readability.py
from cs50 import get_string
import re

# counting charactars


def counting_char(text):
    char = 0
    for i in text:
        if i.islower() or i.isupper():
            char += 1
    return char
    
# counting words


def counting_words(text):
    
    words = text.count(' ') + 1
    return words
    
# counting sentences


def counting_sent(text):
    sent = text.count('.') + text.count('?') + text.count('!')
    return sent 
    

def main():
    # promting the user to enter text
    text = get_string("Text: ")
    # calling the counting functions
    word_sum = counting_words(text)
    alpha_sum = counting_char(text)
    sent_sum = counting_sent(text)
    # calculating Coleman-Liau index  
    index = (((alpha_sum * 100.0) / word_sum) * 0.0588) - (((sent_sum * 100.0) / word_sum) * 0.296) - 15.8
    grad = (round(index))
    
    # check the grad level
    if grad < 1:
        print("Before Grade 1")
    elif grad > 16:
        print("Grade 16+")
    else:
        print("Grade ", grad)


if __name__ == "__main__":
    main()

File: CS50 problem sets/pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int counting_words(string text);
int counting_char(string text);
int counting_sent(string text);

int main(void)
{
    string text = get_string("Text: ");
    
    int wordSum = counting_words(text);
    int alphaSum = counting_char(text);
    int sent = counting_sent(text);
     
    printf("%d \n",wordSum);
    printf("%d \n",alphaSum);
    printf("%d \n",sent);
    //calculation
    float index = (((alphaSum * 100.0) / wordSum) * 0.0588) - (((sent * 100.0) / wordSum) * 0.296) - 15.8;
    int grad = round(index);
    
    if (grad < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grad > 16)
    {
        printf("Grade 16+\n");
    }
    else 
    {
        printf("Grade %i\n", grad);
    }
    
    
}


int counting_words(string text)
{
    int word = 0;
    
    for (int i = 0 ; text[i] != '\0' ; i++)
    {
        if (text[i] == ' ' && text[i + 1] != ' ')
        {
            word++;
        }
        
    }
    word++;
    
    return (word);
}

int counting_char(string text)
{
    int alpha = 0;
    
    for (int i = 0 ; text[i] != '\0' ; i++)
    {
        if (islower(text[i]) || isupper(text[i]))
        {
            alpha++;
        }
    }
    
    return (alpha);
}

int counting_sent(string text)
{
    int sent = 0;
    
    for (int i = 0 ; text[i] != '\0' ; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sent++;
        }
    }
    return (sent);
}